src/AutoRest.CSharp/Common/Output/Models/Types/ModelFactoryTypeProvider.cs

```diff
@@ -5,19 +5,20 @@
 using System.Linq;
 using AutoRest.CSharp.Common.Output.Builders;
 using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Output.Models.Shared;
 
 namespace AutoRest.CSharp.Output.Models.Types
 {
     internal sealed class ModelFactoryTypeProvider : TypeProvider
     {
         protected override string DefaultName { get; }
         protected override string DefaultAccessibility { get; }
-        public IEnumerable<SchemaObjectType> Models { get; }
+        public IEnumerable<MethodSignature> Methods { get; }
         public string DefaultClientName { get; }
 
-        public ModelFactoryTypeProvider(BuildContext context, IEnumerable<SchemaObjectType> models) : base(context)
+        private ModelFactoryTypeProvider(BuildContext context, IEnumerable<MethodSignature> methods) : base(context)
         {
-            Models = models;
+            Methods = methods;
```

Comment:
Should we build methods inside ModelFactoryTypeProvider itlsef?

Reply:
I'm not sure I understand the question.

Reply:
Never mind, I thought we had some logic around factory method creation outside this type but now I see it's right below in a factory method.